toolchain-analysis/bazel.md

```diff
@@ -0,0 +1,41 @@
+# Bazel toolchain analysis
+
+[Bazel](https://bazel.build/) is an open-source build and test tool similar to
+Make, Maven, and Gradle. It uses a human-readable, high-level build language.
+Bazel supports projects in multiple languages and builds outputs for multiple
+platforms. Bazel supports large codebases across multiple repositories, and
+large numbers of users.
+
+## Supported platforms
+
+TODO: test Bazel against currently Node.js supported platforms and toolchains
+(see [BUILDING.MD](https://github.com/nodejs/node/edit/master/BUILDING.md)).
+
+### Unsupported platforms
+
+TODO: see above
+
+## Pros
+
+* Easier to upgrade V8 as we don't need to rewrite build configs for V8
+* Compared to GN, Bazel is more widely adopted outside Chromium
+
+## Cons
+
+* Other dependencies will need Node.js-maintained Bazel build files
```

Comment:
The Bazel team now has a central registry, which includes community-contributed BUILD files for some OSS libraries like zlib and c-ares
https://github.com/bazelbuild/bazel-central-registry/blob/main/modules/c-ares/1.16.1/patches/add_build_file.patch
In practice, I imagine the node project would find a bunch of BUILD files are missing, but at least there's a chance that others will share the maintenance burden.

Reply:
I'll add a note about this registry in the doc, thank you for pointing it out.